dictionary_test.go

```diff
@@ -0,0 +1,118 @@
+package sdk
+
+import (
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDictionaryMethods(t *testing.T) {
+	var str string
+	var genericRes interface{}
+
+	testCases := []struct {
+		Response interface{}
+		Method   string
+		Params   interface{}
+		Callback func(*SDK)
+	}{
+		{
+			// statusLoginRequest
+			Response: &loginResponse{},
+			Method:   "status_login",
+			Params: &statusLoginParam{
+				Address:  "ADDRESS",
+				Password: "PASSWORD",
+			},
+			Callback: func(sdk *SDK) {
+				response, err := statusLoginRequest(sdk, "ADDRESS", "PASSWORD")
+				assert.NoError(t, err)
+				assert.NotNil(t, response)
+			},
+		},
+		{
+			// statusSignupRequest
+			Response: &signupResponse{},
+			Method:   "status_signup",
+			Params: &statusSignupParam{
+				Password: "PASSWORD",
+			},
+			Callback: func(sdk *SDK) {
+				response, err := statusSignupRequest(sdk, "PASSWORD")
+				assert.NoError(t, err)
+				assert.NotNil(t, response)
+			},
+		},
+		{
+			// shhGenerateSymKeyFromPasswordRequest
+			Response: &str,
+			Method:   "shh_generateSymKeyFromPassword",
+			Params:   "PASSWORD",
+			Callback: func(sdk *SDK) {
+				response, err := shhGenerateSymKeyFromPasswordRequest(sdk, "PASSWORD")
+				assert.NoError(t, err)
+				assert.NotNil(t, response)
+			},
+		},
+		{
+			// shhPostRequest
+			Response: &str,
+			Method:   "shh_post",
+			Params:   &Message{},
+			Callback: func(sdk *SDK) {
+				response, err := shhPostRequest(sdk, &Message{})
+				assert.NoError(t, err)
+				assert.NotNil(t, response)
+			},
+		},
+		{
+			// shhGetFilterMessagesRequest
+			Response: &genericRes,
+			Method:   "shh_getFilterMessages",
+			Params:   "",
+			Callback: func(sdk *SDK) {
+				response, err := shhGetFilterMessagesRequest(sdk, "")
+				assert.NoError(t, err)
+				assert.Nil(t, response)
+			},
+		},
+		{
+			// web3Sha3Request
+			Response: &str,
+			Method:   "web3_sha3",
+			Params:   "",
+			Callback: func(sdk *SDK) {
+				response, err := web3Sha3Request(sdk, "")
+				assert.NoError(t, err)
+				assert.NotNil(t, response)
+			},
+		},
+		{
+			// newShhMessageFilterFormatRequest
+			Response: &str,
+			Method:   "shh_newMessageFilter",
+			Params: &shhFilterFormatParam{
+				AllowP2P: true,
+				Topics:   []string{},
+				Type:     "sym",
+				SymKeyID: "",
+			},
+			Callback: func(sdk *SDK) {
+				response, err := newShhMessageFilterFormatRequest(sdk, []string{}, "")
+				assert.NoError(t, err)
+				assert.NotNil(t, response)
+			},
+		},
+	}
+
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	rpcClient := NewMockRPCClient(mockCtrl)
+	sdk := New(rpcClient)
+
+	for _, tc := range testCases {
+		rpcClient.EXPECT().Call(tc.Response, tc.Method, tc.Params)
```

Comment:
Nice! 

Now there is only one problem with this testCases loop: in case one of the tests is broken we won't be able to know which one is it unless we start debugging.

In order to address this we use to do something like:
```go
... 
	testCases := []struct {
                 Description string
		Response interface{}
		Method   string
		Params   interface{}
		Callback func(*SDK)
	}{
		{
                          Description: "statusLoginRequest", // Notice we're adding a field with the description of the current test case
			Response: &loginResponse{},
			Method:   "status_login",
.... 
	for _, tc := range testCases {
		t.Run(tc.Description, func(*testing.T) { // This will, in case a test case is broken, its Description field, so we can skip the debugging phase
		    rpcClient.EXPECT().Call(tc.Response, tc.Method, tc.Params)
		    tc.Callback(sdk)
                 }
	}
```

Reply:
Got it! Will address this and remove the WIP tag